reproducibility_project/src/analysis/diffusion.py

```diff
@@ -0,0 +1,63 @@
+"""Facilitate calculation of the diffusion coefficient from simulation data."""
+
+import gsd.hoomd
+import matplotlib.pyplot as plt
+import numpy as np
+from freud.box import Box
+from freud.msd import MSD
+
+
+def gsd_msd(job, skip=2, stride=1):
+    """Compute the MSD given a Signac Job object.
+
+    The job folder is expected to contain the file "trajectory.gsd" with lengths
+    in nanometers.
+    This function is a convenience wrapper for freud's MSD module
+    https://freud.readthedocs.io/en/latest/modules/msd.html
+    After execution, the files "msd.png" and "msd.txt" are created in the job
+    folder.
+
+    Parameters
+    ----------
+    job : signac.contrib.job.Job
+        The Job object.
+    skip : int, default 2
+        The number of frames from the trajectory to skip.
+    stride : int, default 1
+        The step size between frames.
+
+    Returns
+    -------
+    freud.msd.MSD
+        Computed MSD object
+    """
+    msd = _gsd_msd(job.fn("trajectory.gsd"), skip, stride)
+
+    fig, ax = plt.subplots()
+    ax.plot(msd.msd)
+    ax.set_xlabel("$Frame$")
+    ax.set_ylabel(r"$MSD (nm^{2})$")  # TODO check unit
```

Comment:
Also, if we have `unyt` units associated with these values, we can use [unyt's built-in matplotlib support in a context manager format](https://unyt.readthedocs.io/en/stable/usage.html#plotting-with-matplotlib)

Reply:
wow, that's cool!

Reply:
OK, I did try this, but it got a little wonky with the linear fit.